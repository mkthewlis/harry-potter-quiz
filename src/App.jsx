import './App.css';
import { Quiz } from './Components/Quiz';
import { quizQuestions } from './QuizData/QuizQuestions';
import { randomiseQuizQuestions } from './Calculations/RandomiseQuizQuestions';

function App() {
  const randomQuestions = randomiseQuizQuestions(quizQuestions);
  return (
    <div className="App">
      <Quiz randomQuestions={randomQuestions}/>
    </div>
  );
}

export default App;
