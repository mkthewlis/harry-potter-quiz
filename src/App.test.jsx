import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

jest.mock('./Calculations/RandomiseQuizQuestions', () => ({
  randomiseQuizQuestions: () => ({
    question: 'What do the initials J.K. Rowling stand for?',
    possibleAnswers: [
      { answer: 'Joanne Kathleen', correctAnswer: true },
      { answer: 'Jane Kathleen', correctAnswer: false },
      { answer: 'Jody Katie', correctAnswer: false },
      { answer: 'Joanne Katie', correctAnswer: false }
    ],
  }),
}));

describe('App', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
