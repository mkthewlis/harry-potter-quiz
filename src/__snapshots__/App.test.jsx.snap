// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div
  className="App"
>
  <Quiz
    randomQuestions={
      Object {
        "possibleAnswers": Array [
          Object {
            "answer": "Joanne Kathleen",
            "correctAnswer": true,
          },
          Object {
            "answer": "Jane Kathleen",
            "correctAnswer": false,
          },
          Object {
            "answer": "Jody Katie",
            "correctAnswer": false,
          },
          Object {
            "answer": "Joanne Katie",
            "correctAnswer": false,
          },
        ],
        "question": "What do the initials J.K. Rowling stand for?",
      }
    }
  />
</div>
`;
